lib/contact_screen.dart

```diff
@@ -0,0 +1,317 @@
+      contacts = contacts.take(5).toList();
+    }
+    setState(() {
+      _isLoading = false;
+      contacts.forEach((contactsElement) {
+        final localContact = _contactsList.firstWhereOrNull(
+            (element) => element.name == contactsElement.displayName);
+        if (localContact == null) {
+          _contactsList.add(ContactModel(
+              name: contactsElement.displayName,
+              phoneNumber: contactsElement.phones.isNotEmpty
+                  ? contactsElement.phones[0].number
+                  : ""));
+        }
+      });
+      _searchNotifier.updateSearchValue(_contactsList);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      floatingActionButton: FloatingActionButton.extended(
+        onPressed: () {
+          Navigator.pop(context, _selectedContactsList);
+        },
+        label: const Text('Done'),
+        icon: const Icon(Icons.check, color: Colors.green),
+        backgroundColor: Colors.pink,
+      ),
+      appBar: PreferredSize(
+        preferredSize: const Size(double.infinity, kToolbarHeight),
+        child: ValueListenableBuilder(
+          valueListenable: _valueNotifier,
+          builder: (context, value, child) {
+            return _valueNotifier.value
+                ? appBarForShowOnlyTitle()
+                : appBarForSearch();
+          },
+        ),
+      ),
+      body: _isLoading
+          ? const Center(child: CircularProgressIndicator())
+          : ValueListenableBuilder<List<ContactModel>>(
+              valueListenable: _searchNotifier.contactNotifier,
+              builder: (context, filterList, child) {
+                return ListView.builder(
+                  itemCount: filterList.length,
+                  shrinkWrap: true,
+                  itemBuilder: (BuildContext context, int index) {
+                    return InkWell(
+                      onTap: () {
+                        if (widget.contactSelection ==
+                            ContactSelection.singleSelect) {
+                          Navigator.pop(context, filterList[index]);
+                        } else {
+                          final localContact =
+                              _selectedContactsList.firstWhereOrNull((element) =>
+                                  element.name == filterList[index].name);
+                          if (localContact == null) {
+                            _selectedContactsList.add(filterList[index]);
+                          } else {
+                            _selectedContactsList.remove(localContact);
+                          }
+                          setState(() {});
+                        }
+                      },
+                      child: Column(
+                        children: [
+                          (index > 0 &&
+                                  filterList[index].kilvishId == null &&
+                                  filterList[index].kilvishId != null)
```

Comment:
This condition looks weird .. it will always resolve to false .. `kivlishId == null && kilvishid != null`